packages/js/src/dashboard/widgets/search-ranking-compare-widget.js

```diff
@@ -0,0 +1,196 @@
+			},
+			clicks: {
+				value: data[ 0 ].current.total_clicks || NaN,
+				delta: getDifference( data[ 0 ]?.current?.total_clicks || NaN, data[ 0 ].previous.total_clicks || NaN ),
+			},
+		};
+	}
+
+	return {
+		impressions: {
+			value: data[ 0 ].current.total_impressions || NaN,
+			delta: getDifference( data[ 0 ].current.total_impressions || NaN, data[ 0 ].previous.total_impressions || NaN ),
+		},
+		clicks: {
+			value: data[ 0 ].current.total_clicks || NaN,
+			delta: getDifference( data[ 0 ].current.total_clicks || NaN, data[ 0 ].previous.total_clicks || NaN ),
+		},
+		ctr: {
+			value: data[ 0 ].current.average_ctr || NaN,
+			delta: getDifference( data[ 0 ].current.average_ctr || NaN, data[ 0 ].previous.average_ctr || NaN ),
+		},
+		position: {
+			value: data[ 0 ].current.average_position || NaN,
+			delta: getDifference( data[ 0 ].current.average_position || NaN, data[ 0 ].previous.average_position || NaN ),
+		},
+	};
+};
+
+/**
+ * @param {function} dataTransformer The data transformer.
+ * @param {import("../services/comparison-metrics-data-formatter")} dataFormatter The data formatter.
+ * @returns {function(?TimeBasedData[]): SearchRankingCompareData} Function to format the widget data.
+ */
+export const createIcsaDataFormatter = ( dataTransformer, dataFormatter ) => ( initData ) => {
+	const data = dataTransformer( initData );
+	if ( Object.keys( data ).length === 2 ) {
+		return {
+			impressions: dataFormatter.format( data.impressions, "impressions" ),
+			clicks: dataFormatter.format( data.clicks, "clicks" ),
+		};
+	}
+	return {
+		impressions: dataFormatter.format( data.impressions, "impressions" ),
+		clicks: dataFormatter.format( data.clicks, "clicks" ),
+		ctr: dataFormatter.format( data.ctr, "ctr" ),
+		position: dataFormatter.format( data.position, "position" ),
+	};
+};
+
+/**
+ * @param {import("../services/data-provider")} dataProvider The data provider.
+ * @param {import("../services/remote-data-provider")} remoteDataProvider The remote data provider.
+ * @param {import("../services/comparison-metrics-data-formatter")} dataFormatter The data formatter.
+ * @returns {JSX.Element} The element.
+ */
+export const SearchRankingCompareWidget = ( { dataProvider, remoteDataProvider, dataFormatter } ) => {
+	/**
+	 * @param {RequestInit} options The options.
+	 * @returns {Promise<TimeBasedData|Error>} The promise of IcsaData or an Error.
+	 */
+	const getData = useCallback( ( options ) => {
+		return remoteDataProvider.fetchJson(
+			dataProvider.getEndpoint( "timeBasedSeoMetrics" ),
+			{ options: { widget: "searchRankingCompare" }  },
+			options );
+	}, [ dataProvider ] );
+
+	/**
+	 * @type {function(?TimeBasedData[]): SearchRankingCompareData} Function to format the widget data.
+	 * */
+	const formatIcsaData = useMemo( () => createIcsaDataFormatter( transformData, dataFormatter ), [ transformData, dataFormatter ] );
```

Comment:
About my comment of moving the `transformData` out of the formatter itself. You could combine them here:

```suggestion
	const formatIcsaData = useMemo( () => ( rawData ) => createIcsaDataFormatter( dataFormatter )( transformData( rawData ) ), [ dataFormatter ] );
```